### PR TITLE
support incremental formatting

### PR DESCRIPTION
**Pull request by liuminjian**
This pr resolves https://github.com/opencurve/curveadm/issues/189 .

新增参数--increment
image

### CODE REVIEW

**Comment by liuminjian** [comment]
@caoxianfei1

**Comment by caoxianfei1** [comment]
> @caoxianfei1

got!

**Comment by caoxianfei1** [comment]
@liuminjian squash one commit.

**Comment by liuminjian** [comment]
> @liuminjian squash one commit.

done

File: internal/task/scripts/format.go [hunk 1]
```diff
@@ -28,12 +28,40 @@ percent=$2
...
+else
+    $binary \
+      -allocatePercent=$percent \
+      -fileSize=$chunkfile_size \
+      -filePoolDir=$chunkfile_pool_dir \
+      -filePoolMetaPath=$chunkfile_pool_meta_path \
+      -fileSystemPath=$chunkfile_pool_dir
+fi
+
+
```
Comment (caoxianfei1) [inline internal/task/scripts/format.go:66]: The parameters of curve_format is incremental precent default and seems to be unnecessary to modify the script.

File: internal/task/scripts/format.go [hunk 2]
```diff
@@ -28,12 +28,33 @@ percent=$2
...
+if [ $increment_format == "true" ]
+then
+    rootdir=$(dirname $chunkfile_pool_dir)
+    used_percent=$(df $rootdir --output=pcent|tail -n 1|sed 's/%//'|xargs)
+    let minus=$percent-$used_percent
+    
+    if [ $minus -gt 0 ]
+    then
+      $binary \
```
Comment (Wine93) [inline internal/task/scripts/format.go:43]: Maybe it not works, do you have test it?

Comment (liuminjian) [reply internal/task/scripts/format.go:43]: I have tested it

File: cli/command/format.go [hunk 3]
```diff
@@ -131,6 +134,7 @@ func genFormatPlaybook(curveadm *cli.CurveAdm,
 	stopFormat := options.stopFormat
 	debug := options.debug
 	clean := options.clean
+	increment := options.increment
 
 	steps := FORMAT_PLAYBOOK_STEPS
 	if showStatus {
```
Comment (caoxianfei1) [inline cli/command/format.go:140]: may be we can use playbook `playbook.STOP_SERVICE`.
for example:

```go
 FORMAT_INCREMENT_PLAYBOOK_STEPS := int[]{
  playbook.STOP_SERVICE,
  playbook.FORMAT_CHUNKFILE_POOL,
}

if increment {
  steps = FORMAT_INCREMENT_PLAYBOOK_STEPS
}

stopChunkServercs = curveadm.FilterDeployConfig(dcs, topology.FilterOption{
		Role: "chunkserver",
})
if len(stopChunkServercs) == 0 {
	// no chunkserver
}
if step == playbook.STOP_SERVICE {
     pb.AddStep(&playbook.PlaybookStep{
        ...
        Configs: stopChunkServercs
     } else ...
}
```

Comment (liuminjian) [reply cli/command/format.go:146]: done

**Comment by caoxianfei1** [comment]
LGTM！

**Review by Wine93 — APPROVED** [review]
LGTM!

**Review by caoxianfei1 — APPROVED** [review]

**Pull request merged**